### PR TITLE
133 - setup graphql

### LINKED ISSUE #133: Setup GraphQL
- install gem
- create query `listUser` to list users
- create mutation to `createUser` to create an user
- both needs authentication
- test both at request level

### PR DESCRIPTION
**Pull request by edimossilva**
fix #133

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -65,12 +65,16 @@ gem "sidekiq"
...
+# Ruby implementation of GraphQL [https://github.com/rmosolgo/graphql-ruby]
+gem "graphql"
+
 group :development, :test do
   gem "bullet"
 
   # See https://guides.rubyonrails.org/debugging_rails_applications.html#debugging-with-the-debug-gem
   gem "debug", platforms: %i[mri mingw x64_mingw]
   gem "factory_bot_rails"
+  gem "pry", "~> 0.14.2"
```
Comment (leonardobrito) [inline Gemfile:77]: I think we don't need pry, because we have `binding.irb`, wdyt?

File: Gemfile [hunk 2]
```diff
@@ -102,3 +105,4 @@ group :test do
   gem "simplecov", require: false
   gem "spring-commands-rspec"
 end
+gem "graphiql-rails", group: :development
```
Comment (leonardobrito) [inline Gemfile:108]: Can you move to above☝🏻?
```suggestion
gem "graphiql-rails"
```

File: .reek.yml [hunk 3]
```diff
@@ -81,3 +81,4 @@ exclude_paths:
   - bin/
   - vendor/
   - db/migrate
+  - app/controllers/graphql_controller.rb
```
Comment (leonardobrito) [inline .reek.yml:84]: Why are we excluding?

File: spec/requests/graphql/users_query_spec.rb [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+    let(:query_string) do
+      "{
+        users {
+          id
+          email
+        }
+      }"
+    end
+
+    let(:request) { post graphql_path, params: { query: query_string } }
```
Comment (leonardobrito) [inline spec/requests/graphql/users_query_spec.rb:25]: Sounds good use subject?
```suggestion
    subject { post graphql_path, params: { query: query_string } }
```

File: app/graphql/types/mutation_type.rb [hunk 5]
```diff
@@ -0,0 +1,12 @@
+# frozen_string_literal: true
+
+module Types
+  class MutationType < Types::BaseObject
+    # TODO: remove me
```
Comment (leonardobrito) [inline app/graphql/types/mutation_type.rb:5]: 🤔

File: spec/requests/graphql/users_query_spec.rb [hunk 6]
```diff
@@ -0,0 +1,34 @@
+# frozen_string_literal: true
+
+require "rails_helper"
+
+RSpec.describe "users query" do
```
Comment (leonardobrito) [inline spec/requests/graphql/users_query_spec.rb:5]: Can you add an example of mutation? 👀

File: Gemfile (hunk 1 above)
Comment (edimossilva) [reply Gemfile:77]: personal preference

File: .reek.yml (hunk 3 above)
Comment (edimossilva) [reply .reek.yml:84]: generated file, from graphql gem, I dont want to refactor it to meet reek criterias

File: spec/requests/graphql/users_query_spec.rb (hunk 4 above)
Comment (edimossilva) [reply spec/requests/graphql/users_query_spec.rb:25]: rubocop complains if we do it

File: spec/requests/graphql/users_query_spec.rb (hunk 6 above)
Comment (edimossilva) [reply spec/requests/graphql/users_query_spec.rb:5]: issue was created for this https://github.com/espoo-dev/rails_boilerplate/issues/146

File: app/graphql/types/mutation_type.rb (hunk 5 above)
Comment (edimossilva) [reply app/graphql/types/mutation_type.rb:5]: will be fixed by https://github.com/espoo-dev/rails_boilerplate/issues/146

File: Gemfile [hunk 7]
```diff
@@ -102,3 +105,4 @@ group :test do
   gem "simplecov", require: false
   gem "spring-commands-rspec"
 end
+gem "graphiql-rails"
```
Comment (edimossilva) [inline Gemfile:108]:
```suggestion
```

File: Gemfile [hunk 8]
```diff
@@ -102,3 +105,4 @@ group :test do
   gem "simplecov", require: false
   gem "spring-commands-rspec"
```
Comment (edimossilva) [inline Gemfile:106]:
```suggestion
  gem "spring-commands-rspec"
  gem "graphiql-rails"
```

**Pull request merged**